backend/dataall/__init__.py

```diff
@@ -1,2 +1,13 @@
 from . import core, version
 from .base import utils, db, api
+import logging
+import os
+import sys
+
+logging.basicConfig(
```

Comment:
If I understand correctly this statement here will affect all data.all backend services (lambda, ecs tasks etc).
Should we be removing ALL  the per file `logging` configs? With a quip grep I see 206 files
`grep -rail "logging.getLogger(" backend/ | wc -l`

Reply:
You are correct it affects every file under `/backend/dataall/` - I though this would simplify how we were formerly managing logs in each individual location 

It also solves the issue we have where a task (like share manager task - at `dataall/backend/dataall/modules/shares_base/tasks/share_manager_task.py` ) was not recording the logs from SharingService (at `dataall/backend/dataall/modules/shares_base/services/sharing_service.py`) or the Processors and we were missing logs in CloudWatch


For Reference in `dataall-sbx-backend-graphql` the log group formatting before the change:
```
[INFO]	2024-11-19T15:28:43.058Z	a725615b-f975-4a66-8ccb-7e69830f18b9	Current maintenance window status - INACTIVE
[INFO]	2024-11-19T15:28:43.060Z	a725615b-f975-4a66-8ccb-7e69830f18b9	SSM Parameter session in central account
```

And after the code change (same formatting):

```
[INFO]	2024-11-20T22:34:02.450Z	2133b99e-3316-44de-9b9c-68f02f96611c	Current maintenance window status - INACTIVE
[INFO]	2024-11-20T22:34:02.452Z	2133b99e-3316-44de-9b9c-68f02f96611c	SSM Parameter session in central account
```

Reply:
To note the above differs from the `format='[%(levelname)s] %(message)s',`  structure. I believe Lambda has its own default logging formatter that is taking precedence... this log record format of `[%(levelname)s] %(message)s',` is following in all of the ECS tasks for example:

 
![Screenshot 2024-11-21 at 9 17 40 AM](https://github.com/user-attachments/assets/3e4e6616-3b55-465a-b97f-46b284e6bfe0)

Reply:
@dlpzx we do configure at a lot of places
@noah-paige I am fine if you want to refactor and remove the configs from all the files keeping only the top one

Reply:
For a full breakdown of where we call `logger.setLevel(...)` which I think is the main config we do in a number of places (was ~26 places before this PR now at 15 files):

- Backend Files not under `dataall/backend/dataall/` (5):
    - dataall/backend/api_handler.py
    - dataall/backend/aws_handler.py
    - dataall/backend/local_graphql_server.py
    - dataall/backend/search_handler.py
- For rest of Backend Code:
    - dataall/backend/dataall/__init__.py
- For CDK specific activity:
    - dataall/backend/dataall/base/cdkproxy/app.py (creates new logger named `cdkapp process` always with level INFO) -- chose to leave as is
- Lambdas from data.all resource Custom Resource CDK
    - dataall/backend/dataall/modules/s3_datasets/cdk/assets/gluedatabasecustomresource/index.py
    - dataall/backend/dataall/modules/s3_datasets/cdk/assets/lakeformationdefaultsettings/index.py
- Trigger Function Lambdas data.all pipeline
    - dataall/backend/deployment_triggers/dbmigrations_handler.py
    - dataall/backend/deployment_triggers/dbsnapshots_handler.py
    - dataall/backend/deployment_triggers/saveperms_handler.py
- Custom Resource Lambdas data.all Deployment
    - dataall/deploy/custom_resources/cognito_config/cognito_urls.py
    - dataall/deploy/custom_resources/cognito_config/cognito_users.py
    - dataall/deploy/custom_resources/custom_authorizer/custom_authorizer_lambda.py
    - dataall/deploy/custom_resources/custom_authorizer/jwt_services.py

Reply:
@dlpzx @petrkalos - the places where we configure logs is more intentional now and for the majority of backend it is from `dataall/backend/dataall/__init__.py` which I think is best

Some that are in different compute functions or parts of deployment will remain separate